src/server-renderer.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { renderToString } from 'react-dom/server';
+import { StaticRouter } from 'react-router';
+import Invision from './invision/invision';
+import { store } from './invision/store';
```

Comment:
I would reccomend you create **configureStore** which may take an initial value as a parameter. After that in **App** component you can create store with store value from the server (**window.PRELOADED_STATE**) and create **Invision** componet with current store value